engine/src/flutter/shell/platform/darwin/ios/framework/Source/TextRange.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import UIKit
+
+/// A UITextPosition subclass that represents a location in the text, in
+/// a Flutter text field.
+@objc(FlutterTextPosition)
+final class TextPosition: UITextPosition, NSCopying {
+  // offset is exposed as `range` in an extension on UITextPosition.
+  fileprivate let offset: UInt
+  @objc let affinity: UITextStorageDirection
+
+  init(index: UInt, affinity: UITextStorageDirection = .forward) {
+    self.offset = index
+    self.affinity = affinity
+  }
+
+  override func isEqual(_ object: Any?) -> Bool {
+    guard let other = object as? TextPosition else {
+      return false
+    }
+    // Affinity is only visual.
```

Comment:
This comment doesn't explain how something "only visual" is equivalent to "don't consider it in isEqual". I wouldn't be surprised if apple writes code like this: 

```
updateVisualIfNeeded(position: Position) {
  if position == self.position { return }
  // update visual
}
```

Reply:
removed the override since it wasn't in the objc version.